src/lib/formats/ap2_dsk.cpp

```diff
@@ -313,7 +313,9 @@ bool a2_16sect_format::load(util::random_read &io, uint32_t form_factor, const s
 		uint8_t sector_data[APPLE2_SECTOR_SIZE*16];
 
 		auto const [err, actual] = read_at(io, fpos, sector_data, sizeof sector_data);
-		if (err || actual != sizeof sector_data)
+		// Some supported images have oddball sizes, where the last track is incomplete.
+		// Skip the `actual` check to avoid rejecting them.
+		if (err /* || actual != sizeof sector_data */)
 			return false;
```

Comment:
Isn’t this going to behave non-deterministically on these images with incomplete final tracks, because the sector data buffer is uninitialised?

What do these images represent?  Disks with fewer than the standard number of sectors on the final track?  Truncated sectors implicitly zero-filled?  Something else?

Should it be checking that actual length read is a whole number of sectors?  Should it only allow the read to return short on the final track?  Should it generate a track with fewer than the standard number of sectors if the read returns short?

Reply:
> Isn’t this going to behave non-deterministically on these images with incomplete final tracks, because the sector data buffer is uninitialised?

Yeah, but that's the way it was before 47d8c69b. I just reverted to the original behavior.

> What do these images represent? Disks with fewer than the standard number of sectors on the final track? Truncated sectors implicitly zero-filled? Something else?

My best guess is that they are simply incorrectly truncated images. I could later double-check with the woz images whether the missing bytes are actually zeroes or if any data was lost.

> Should it be checking that actual length read is a whole number of sectors? Should it only allow the read to return short on the final track? Should it generate a track with fewer than the standard number of sectors if the read returns short?

Personally, I think that `bardstl3` should be removed from the list, given that its images are malformed, and there's a better replacement (`brdtale3`). Then the check can be restored. But I don't know what the MAME policy is for such cases.